programs/mango-v4/src/accounts_ix/openbook_v2_place_order.rs

```diff
@@ -58,17 +58,29 @@ pub struct OpenbookV2PlaceOrder<'info> {
     /// CHECK: Validated by the openbook_v2 cpi call
     pub market_vault_signer: UncheckedAccount<'info>,
 
-    /// The bank that pays for the order, if necessary
+    /// The bank that pays for the order
     // token_index and payer_bank.vault == payer_vault is validated inline at #3
     #[account(mut, has_one = group)]
-    pub bank: AccountLoader<'info, Bank>,
-    /// The bank vault that pays for the order, if necessary
+    pub payer_bank: AccountLoader<'info, Bank>,
+    /// The bank vault that pays for the order
     #[account(mut)]
-    pub vault: Box<Account<'info, TokenAccount>>,
+    pub payer_vault: Box<Account<'info, TokenAccount>>,
+
+    /// The bank that receives the funds upon settlement
+    // token_index and payer_bank.vault == payer_vault is validated inline at #3
+    #[account(mut, has_one = group)]
+    pub receiver_bank: AccountLoader<'info, Bank>,
+    /// The bank vault that pays for the order
+    #[account(mut)]
+    pub receiver_vault: Box<Account<'info, TokenAccount>>,
+
+    /// CHECK: The oracle can be one of several different account types
+    #[account(address = payer_bank.load()?.oracle)]
+    pub payer_oracle: UncheckedAccount<'info>,
 
     /// CHECK: The oracle can be one of several different account types
-    #[account(address = bank.load()?.oracle)]
-    pub oracle: UncheckedAccount<'info>,
+    #[account(address = receiver_bank.load()?.oracle)]
+    pub receiver_oracle: UncheckedAccount<'info>,
```

Comment:
Sorry for not being clearer, imo:
- explicitly pass the payer_bank and receiver_bank, because they need to be writable - and normally health accounts in remaining_accounts are not writable
- the receiver_vault is not needed and does not need to be passed
- it's more future proof to read the oracle prices from the health accounts with the account_retriever (like you are doing below), passing them explicitly here will need to be removed in the future anyway, so better drop receiver_oracle and payer_oracle from the explicit account list